### DISCUSSION TITLE
Why did you not use BSL?

### DISCUSSION THREAD

**Discussion by hhimanshu** [plausible/analytics#1195] [Anything Else]
Hello folks!

I love and use your product, I paid for your service and believe in what you are doing. I am also learning about open-source licensing which works in alignment with monetisation.
Earlier I read https://plausible.io/blog/open-source-licenses and found it to be impressive and right move.

Today, I watched [Business Source Licensing](https://www.youtube.com/watch?v=dgLzRuG_Ty8) presentation which made me think that this is a great license to have. This will encourage innovation while helping to create a self-sustaining business.

So, I am reaching out to find out whether/if you considered BSL when evaluating your licenses. And if you did, what features of [GNU Affero General Public License v3.0](https://github.com/plausible/analytics/blob/master/LICENSE.md) did you folks found appealing over BSL for Plausible?

Do you mind educating me?
Either way, I want open-source businesses to flourish and want to learn from them.

I hope to learn more from you
Thank you

**Comment by metmarkosaric** [comment]
thank you for your support @hhimanshu! didn't consider BSL so don't really have much opinion on it. our thought process, reasoning and situation was pretty much explained in the blog post you linked to. what more does it do than AGPL?

**Comment by ukutaht** [comment]
I found out about BSL after we'd made the switch and honestly it looks great. I'm sure we would have considered it for this project had we known about it and I plan to seriously consider it for any future projects.

@metmarkosaric BSL kinda works like the patent law. As the author you get exclusive rights for commercial use for a *limited period of time*.  The code is open but the license restricts others from using it for commercial purposes, usually for 4 years. After that the project enters the public domain as open-source and everybody has rights to commercialize it.

It's not considered strictly open-source but more like *'eventually open source'*

**Reply by metmarkosaric** [reply]
thanks for the details! the issue with the licenses that are not considered fully open source is that you won't get as much community involvement / buzz. and community involvement is one great reason to go open source. so AGPL is better for that reason alone. also since switching to AGPL, the "competitors taking our code" problem has kind of gone away so AGPL does help with that too.

**Reply by sumpfralle** [reply]
I empathize with your wish of preventing competitors from exploiting your work.
The AGPL is just the right tool for this goal.

Regarding your statement on the BSL:

> As the author you get exclusive rights for commercial [..]

The definition of *commercial* is (at least here in Germany) quite broad and not as precise, as one could think. For example websites, which try to sell stuff or show any kind of advertisement, affiliate link or other venues of income are considered *commercial*. Thus the self-hosted plausible would be usable only for a tiny fraction of websites out there.
(even if the above description of *commercial* may be disputable in many legislations, the mere risk will surely scare users away)

And (as others pointed out before): using a non-free license prevents plausible from being distributed by Debian and other distributions. This will significantly raise the threshold of using it for many users/administrators out there.

Last but least: the AGPL transmits a clear statement of *we will not exploit our users* - not just as a vague promise, but as a binding contract.
(personally, I would not even have bothered to look at *plausible*, if it would have used any "weird" license - there are just too many other wild SaaS competitors with crazy monetization schemes out there ...)

**Comment by kmohrf** [comment]
Just wanted to weigh in and say that I’m glad you chose the AGPL to tackle this problem. As someone who’s involved in a lot of small and local free software projects I must admit that I’m usually not affected by the same financial constraints you seem to have and I can honestly say that I never had to compete with Big-Tech. I’ve used the AGPL myself and I’m eager to see if it will help you in this regard because the BSL feels like a big step backwards for me. It traps linux distributions in an impossible situation where they’re demoted to simple library-bundlers, unable to package any relevant software applications – at least in a timeframe that is meaningful to the larger software ecosystem. I also don’t see how security-fix backports to older releases may happen given the fact that backporting code in general seems to be a no-go with the BSL. I doubt that the software companies that adopt the BSL will publish security fixes for their 3-4 year old products that have reached the license change date and I guess we’ll find ourselves in a situation where it becomes the norm to run-from-latest-or-don’t-run-at-all, making the license change basically irrelevant.

So… thanks for taking the leap of faith to embrace the AGPL with it’s open nature! I honestly hope that you fare well with it and wish you all the best! And for what it’s worth: I do hope we find other methods to tackle predatory Big-Tech, like… taxing their buttocks and smashing cartels for a change :).

**Comment by hhimanshu** [comment]
Thank you for the great discussion, I almost missed this education. Thank you, everyone, genuinely for sharing your thoughts
